src/error.rs

```diff
@@ -32,4 +45,56 @@ pub enum DatabaseError {
     WriteError(String),
     #[error("Failed to delete {0} from the Database database")]
     DeleteError(String),
-}
\ No newline at end of file
+}
+
+#[derive(Error, Debug)]
+pub enum DataAvailabilityError {
+    #[error("Failed to establish WebSocket connection: {0}")]
+    WebSocketError(String),
+    #[error("Namespace initialization error")]
+    NamespaceInitializationError,
```

Comment:
Namespace is something Celestia specific and is also more of a config error than anything, so just keeping this in mind in case we come up with anything better... maybe leave a todo comment? or do you have an idea? idk

Reply:
I see... What do you think if we use one of the already existing but parameterizable errors? InitializationError, ConnectionError or NetworkError? We can / should adapt the custom message.

Reply:
InitializationError works great for this